File: src/library/Shift.Sdk/Contract/Evaluation/Workshops/Models/WorkshopAttachment.cs
using System;

using Shift.Constant;

namespace Shift.Contract
{
    public class WorkshopAttachment
    {
        public Guid AttachmentId { get; set; }
        public AttachmentType AttachmentType { get; set; }
        public int AssetNumber { get; set; }
        public int AssetVersion { get; set; }
        public string Title { get; set; }
        public string Condition { get; set; }
        public string PublicationStatus { get; set; }
        public int QuestionCount { get; set; }
        public DateTimeOffset PostedOn { get; set; }
        public string FileName { get; set; }
        public string FileUrl { get; set; }
        public string FileSize { get; set; }
        public string AuthorName { get; set; }
        public int ChangeCount { get; set; }

        // Image-specific
        public string ImageResolution { get; set; }
        public string[] ImageDimensions { get; set; }
        public string Color { get; set; }
    }
}

File: src/ui/Shift.UI/src/contexts/workshop/models/WorkshopAttachment.ts
import { DateTimeParts } from "@/helpers/date/dateTimeTypes";

export type AttachmentType = "Unknown" | "Archive" | "Document" | "Image";

export interface WorkshopAttachment {
    attachmentId: string;
    attachmentType: AttachmentType;
    assetNumber: number;
    assetVersion: number;
    title: string | null;
    condition: string | null;
    publicationStatus: string;
    questionCount: number;
    postedOn: DateTimeParts;
    fileName: string | null;
    fileUrl: string | null;
    fileSize: string | null;
    authorName: string | null;
    changeCount: number;
    imageResolution: string | null;
    imageDimensions: string[] | null;
    color: string | null;
}